lib/client.js

```diff
@@ -327,6 +334,65 @@ class Client extends EventEmitter {
     return !this.full
   }
 
+  pipeline (opts, handler) {
+    if (typeof handler !== 'function') {
+      throw new Error('invalid handler')
+    }
+
+    const req = new PassThrough({ autoDestroy: true })
+    const res = new PassThrough({ autoDestroy: true })
+    const ret = new Duplex({
+      autoDestroy: true,
+      read () {
+        res.resume()
+
+        // TODO: Workaround for possible Node bug?
+        // Doesn't always emit 'end' if resumed after
+        // ended.
```

Comment:
@mcollina: FYI. I haven't had time to dig into this yet but it seems like a Node core bug affecting all versions of Node I've tested (12-14).

Reply:
I can reproduce on Node master.

Reply:
Found it. Not a core bug.

Reply:
https://github.com/mcollina/undici/pull/150